### ISSUE TITLE
bug: visual highlights get stuck

### ISSUE THREAD

**Issue by alivault** [LazyVim/LazyVim#3850]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.10.0

### Operating system/version

MacOS 14.4.1 

### Describe the bug

Visual highlights keep getting stuck. Happens often. I have to close and reopen the buffer to clear it. But when I reopen the buffer and begin to make changes, I get an error `Error executing lua callback: ... lar/neovim/0.10.0/share/nvim/runtime/lua/vim/snippet.lua:101: attempt to index a nil value`

Video of the stuck visual highlight:

https://github.com/LazyVim/LazyVim/assets/48610822/762e697b-b908-4189-b425-71a769414d0c

Screenshot of errors once I close the buffer, reopen and begin to make changes to the buffer:
<img width="1728" alt="Screenshot 2024-06-28 at 7 46 09 PM" src="https://github.com/LazyVim/LazyVim/assets/48610822/b282139a-2fdb-4bc2-a579-c8950ee008c4">

### Steps To Reproduce

Happens often when I use this pattern to wrap some lines with something like a div or brackets:

1. Highlight some lines using `<S-v>`
2. Change the lines using `c`
3. Type in something that auto-closes like brackets `(` or some tag using emmet like `<div>`
4. Now that your cursor is in the between the auto-closed pair, paste in the lines you just removed by using your system's paste command

I can't guarentee that this will cause the stuck highlights, but as I'm working quickly, I often get see this bug at least once a day.

### Expected Behavior

Visual highlights should not get stuck and there should be no error `Error executing lua callback: ... lar/neovim/0.10.0/share/nvim/runtime/lua/vim/snippet.lua:101: attempt to index a nil value`

### Repro

```Lua
default init.lua
```

**Comment by folke** [comment]
I noticed that too. Seems to be caused by Neovim builtin snippets. Please file an issue there (or look for an existing one, which I'm sure already exists)

**Comment by alivault** [comment]
Thank you for your prompt response. I filed an issue at neovim. I also wanted to share that I found a quick solution to the issue. By installing the `luasnip` extra, the bugs seem to go away. This might be a helpful workaround for others experiencing the same problem. It also solved another bug when creating a new file and then using a snippet like `rfcv` to generate a component, the name of the file would not be included with the snippet. Now with `luasnip`, it works as intended.
